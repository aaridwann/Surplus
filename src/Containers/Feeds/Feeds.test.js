import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import FeedsComponent from './Feeds';

const data = [
	{
		picture: {
			large: 'https://example.com/image1.jpg',
		},
		name: {
			first: 'John',
		},
	},
	{
		picture: {
			large: 'https://example.com/image2.jpg',
		},
		name: {
			first: 'Jane',
		},
	},
];

describe('FeedsComponent', () => {
	it('should render the component correctly', () => {
		const onPressMock = jest.fn();
		const { getByTestId, getAllByTestId } = render(
			<FeedsComponent data={data} onPress={onPressMock} />
		);

		const feedsComponent = getByTestId('feeds-component');
		expect(feedsComponent).toBeDefined();

		const feedNames = getAllByTestId('feed-name');
		expect(feedNames).toHaveLength(data.length);

		fireEvent.press(feedNames[0]);
		expect(onPressMock).toHaveBeenCalledWith(data[0]);
	});

	it('should match the snapshot', () => {
		const onPressMock = jest.fn();
		const { toJSON } = render(
			<FeedsComponent data={data} onPress={onPressMock} />
		);

		expect(toJSON()).toMatchSnapshot();
	});
});
