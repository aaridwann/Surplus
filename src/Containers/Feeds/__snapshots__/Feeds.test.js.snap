// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedsComponent should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
  testID="feeds-component"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#50b8a8",
        "height": 200,
        "justifyContent": "center",
        "margin": 5,
        "overflow": "hidden",
        "width": 340,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://example.com/image1.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "margin": 4,
          "width": "100%",
        }
      }
    />
    <Text
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "flex-start",
          "bottom": 0,
          "color": "white",
          "fontSize": 60,
          "justifyContent": "flex-start",
          "margin": 4,
          "position": "absolute",
          "textTransform": "capitalize",
        }
      }
      testID="feed-name"
    >
      John
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#50b8a8",
        "height": 200,
        "justifyContent": "center",
        "margin": 5,
        "overflow": "hidden",
        "width": 340,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://example.com/image2.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "margin": 4,
          "width": "100%",
        }
      }
    />
    <Text
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "flex-start",
          "bottom": 0,
          "color": "white",
          "fontSize": 60,
          "justifyContent": "flex-start",
          "margin": 4,
          "position": "absolute",
          "textTransform": "capitalize",
        }
      }
      testID="feed-name"
    >
      Jane
    </Text>
  </View>
</View>
`;
